flutter_flexible_sync/lib/main.dart

```diff
@@ -41,6 +41,7 @@ Future<Realm> createRealm(String appId, CollectionType collectionType) async {
   });
 
   await realm.subscriptions.waitForSynchronization();
+  await realm.syncSession.waitForDownload();
```

Comment:
Awaiting these potentially infinite calls in main is not a good idea.

Reply:
I will revert this change and will create another task to restructure the samples.